examples/session_sharing.ipynb

```diff
@@ -0,0 +1,585 @@
+    "\n",
+    "This package is implemented in a way that it automatically creates, caches, and reuses an authentication session within a single Python runtime process. Once the current session expires or the runtime process is restarted it will automatically create a new session. This way normal users don't have to worry about interacting with session objects directly. However, once users parallelize their processes over multiple Python processes or even over a cluster of multiple compute instances then they should make sure that a single authentication session is shared between all of them.\n",
+    "\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "As an example, let's say a user splits their area of interest into `10.000` smaller bounding boxes. Then they parallelize data download from Sentinel Hub over these bounding boxes using `10` different Python processes. It depends on the type of parallelization, but without implementing a session sharing mechanism such a procedure can create `10.000` different authentication sessions. This can cause the user to be temporary restricted from using Sentinel Hub services. In such cases please contact [support](https://docs.sentinel-hub.com/api/latest/support/).\n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "This tutorial will explain the basic mechanism of working with sessions in `sentinelhub-py` and how to implement a session transfer for different types of parallelization. \n",
+    "\n",
+    "For the purposes of the tutorial, let's first configure a logging handler that will tell us exactly when a new session has been created:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "grand-fossil",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import sys\n",
+    "from logging import Filter, Formatter, LogRecord, StreamHandler\n",
+    "\n",
+    "\n",
+    "class SessionFilter(Filter):\n",
+    "    \"\"\"A filter that removes every log record except the ones coming from\n",
+    "    sentinelhub.download.session module.\"\"\"\n",
+    "\n",
+    "    def filter(self, record: LogRecord) -> bool:\n",
+    "        return record.name == \"sentinelhub.download.session\"\n",
+    "\n",
+    "\n",
+    "handler = StreamHandler(sys.stdout)\n",
+    "handler.setLevel(logging.DEBUG)\n",
+    "\n",
+    "formatter = Formatter(\"%(asctime)s: %(message)s\")\n",
+    "handler.setFormatter(formatter)\n",
+    "\n",
+    "log_filter = SessionFilter()\n",
+    "handler.addFilter(log_filter)\n",
+    "\n",
+    "global_logger = logging.getLogger()\n",
+    "global_logger.setLevel(logging.DEBUG)\n",
+    "\n",
+    "for handler in global_logger.handlers:\n",
+    "    global_logger.removeHandler(handler)\n",
```

Comment:
Yes, it removes the default handler because we have to set the default global logger to debug.

Actually, now I got an idea how to write this a bit differently. I'll try it out.

Reply:
It worked! Logging setup code is now much simpler.